inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/reader/external/IncrementalSourceStreamFetcher.java

```diff
@@ -97,25 +100,27 @@ public void submitTask(FetchTask<SourceSplitBase> fetchTask) {
 
     @Override
     public boolean isFinished() {
-        return currentStreamSplit == null || !streamFetchTask.isRunning();
+        return currentStreamSplit == null || !currentTaskRunning;
     }
 
     @Nullable
     @Override
     public Iterator<SourceRecords> pollSplitRecords() throws InterruptedException {
         checkReadException();
         final List<SourceRecord> sourceRecords = new ArrayList<>();
-        if (streamFetchTask.isRunning()) {
+        if (currentTaskRunning) {
```

Comment:
Why replace `streamFetchTask.isRunning()`?

Reply:
You can implement  `PostgresIncrementalSourceStreamFetcher` class by inheriting from `IncrementalSourceStreamFetcher` instead of modifying the logic in `IncrementalSourceStreamFetcher`. You can refer to https://github.com/apache/inlong/blob/master/inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/source/reader/fetch/MongoDBScanFetchTask.java.